runtime/runtime/src/actions.rs

```diff
@@ -397,11 +399,24 @@ pub(crate) fn try_refund_allowance(
 
 pub(crate) fn action_transfer(
     account: &mut Account,
-    transfer: &TransferAction,
+    deposit: Balance,
+    nonrefundable: bool,
```

Comment:
Instead of having a boolean flag here it probably is better to have two separate functions: the original `action_transfer` and a new function `action_nonrefundable_transfer`. This better reflects the action model (there is a separate action now instead of a flag on the transfer action), and reduces complexity in the code by having less branching.